   Compiling c2rust_out v0.0.0 (/home/c00834010/dev/crusts-benchmarks/2023-06-27/apply_a_callback_to_an_array)
error: default numeric fallback might occur
  --> src/main.rs:24:13
   |
24 |         i = 0;
   |             ^ help: consider adding suffix: `0_i32`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_numeric_fallback
   = note: requested on the command line with `-D clippy::default-numeric-fallback`

error: unsafe block missing a safety comment
  --> src/main.rs:22:5
   |
22 |     unsafe {
   |     ^^^^^^^^
   |
   = help: consider adding a safety comment on the preceding line
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#undocumented_unsafe_blocks
   = note: requested on the command line with `-D clippy::undocumented-unsafe-blocks`

error: used `expect()` on an `Option` value
  --> src/main.rs:26:13
   |
26 |             callback.expect("non-null function pointer")(i, *array.offset(i as isize));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: if this value is `None`, it will panic
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_used
   = note: requested on the command line with `-D clippy::expect-used`

error: default numeric fallback might occur
  --> src/main.rs:40:12
   |
40 |     return 0;
   |            ^ help: consider adding suffix: `0_i32`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#default_numeric_fallback

error: could not compile `c2rust_out` (bin "main") due to 4 previous errors
